externs/browser/streamsapi.js

```diff
@@ -14,294 +14,446 @@
  * limitations under the License.
  */
 /**
- * @fileoverview Definitions from the Streams API as of 22 June 2015.
- * @see https://streams.spec.whatwg.org/commit-snapshots/b2131ef6460c446817767d7575698f9fcfa554a3/
+ * @fileoverview Streams API definitions
+ * 
+ * Based on Living Standard — Last Updated 5 August 2016
+ * https://streams.spec.whatwg.org/commit-snapshots/34ecaadbcce8df9943d7a2cdb7fca4dc25914df4/
+ * 
+ * @see https://streams.spec.whatwg.org/
  * @externs
  */
 
+
+/** @typedef {{ value:*, done:boolean }} */
+var IteratorResult;
+
+
 /**
- * @typedef {{ value: (*|undefined), done: boolean }}
+ * @typedef {!CountQueuingStrategy|!ByteLengthQueuingStrategy|{
+ *     size: (undefined|function(*): number),
+ *     highWaterMark: number
+ * }}
  */
-var IteratorResult;
+var QueuingStrategy;
+
 
 /**
- * @param {{ highWaterMark: number }} config
- * @constructor
+ * @record
  */
-function ByteLengthQueuingStrategy(config) {}
+function TransformStream() {};
+
+/** @type {!WritableStream} */
+TransformStream.prototype.writable;
+
+/** @type {!ReadableStream} */
+TransformStream.prototype.readable;
+
 
 /**
- * @param {*} chunk
- * @return {number}
+ * @record
  */
-ByteLengthQueuingStrategy.prototype.size = function(chunk) {};
+function PipeOptions() {};
+
+/** @type {boolean} */
+PipeOptions.prototype.preventClose;
+
+/** @type {boolean} */
+PipeOptions.prototype.preventAbort;
+
+/** @type {boolean} */
+PipeOptions.prototype.preventCancel;
+
+
+/**
+ * @record
+ */
+function ReadableStreamSource() {};
+
+/** @type {(undefined|function((!ReadableByteStreamController|!ReadableStreamDefaultController)):(!Promise<*>|undefined))}*/
+ReadableStreamSource.prototype.start;
```

Comment:
I would like to change this and `ReadableStreamSource.prototype.pull;` to 
```
{(undefined
|function(!ReadableByteStreamController):(!Promise<*>|undefined)
|function(!ReadableStreamDefaultController):(!Promise<*>|undefined))}
```
because currently I have to deal with `!ReadableByteStreamController|!ReadableStreamDefaultController` type inside these methods.

What do you think @MatrixFrog ? Will it be OK if I update this PR? Or would be better to leave it for another PR after this one is merged?